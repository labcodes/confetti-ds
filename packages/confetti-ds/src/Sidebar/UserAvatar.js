import React from "react";
import PropTypes from "prop-types";

export default class UserAvatar extends React.Component {
  static propTypes = {
    /** Source of the avatar to be rendered. */
    avatarSrc: PropTypes.string.isRequired,
    /** Text that will specify the HTML alt attribute of an <img> element. */
    altText: PropTypes.string.isRequired,
    /** Text that will be rendered as the avatar's caption. */
    caption: PropTypes.string.isRequired,
  };

  render() {
    const { avatarSrc, altText, caption } = this.props;
    return (
      <div className="lab-narrow-sidebar__avatar">
        <div className="lab-narrow-sidebar__avatar-photo">
          <img src={avatarSrc} alt={altText} />
        </div>
        <div className="lab-narrow-sidebar__avatar-text">
          <span>{altText}</span>
          <span className="lab-narrow-sidebar__avatar-caption">{caption}</span>
        </div>
      </div>
    );
  }
}
