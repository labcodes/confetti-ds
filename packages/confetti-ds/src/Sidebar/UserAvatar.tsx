import React from "react";

interface UserAvatarProps {
  /** Source of the avatar to be rendered. */
  avatarSrc: string,
  /** Text that will specify the HTML alt attribute of an <img> element. */
  altText: string,
  /** Text that will be rendered as the avatar's caption. */
  caption: string,
};

export default function UserAvatar({ avatarSrc, altText, caption }: UserAvatarProps) {
  return (
    <div className="lab-narrow-sidebar__avatar">
      <div className="lab-narrow-sidebar__avatar-photo">
        <img src={avatarSrc} alt={altText} />
      </div>
      <div className="lab-narrow-sidebar__avatar-text">
        <span>{altText}</span>
        <span className="lab-narrow-sidebar__avatar-caption">{caption}</span>
      </div>
    </div>
  );
}
