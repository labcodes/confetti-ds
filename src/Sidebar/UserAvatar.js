import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param avatarSrc Source of the avatar to be rendered.
 * @param altText Text that will specify the HTML alt attribute of an <img> element.
 * @param caption Text that will be rendered as the avatar's caption.
 * @returns {JSX.Element}
 * @constructor
 */
export default function UserAvatar({ avatarSrc, altText, caption }) {
  return (
    <div className="lab-narrow-sidebar__avatar">
      <div className="lab-narrow-sidebar__avatar-photo">
        <img src={avatarSrc} alt={altText} />
      </div>
      <div className="lab-narrow-sidebar__avatar-text">
        <span>{altText}</span>
        <span className="lab-narrow-sidebar__avatar-caption">{caption}</span>
      </div>
    </div>
  );
}

UserAvatar.propTypes = {
  /** Source of the avatar to be rendered. */
  avatarSrc: PropTypes.string.isRequired,
  /** Text that will specify the HTML alt attribute of an <img> element. */
  altText: PropTypes.string.isRequired,
  /** Text that will be rendered as the avatar's caption. */
  caption: PropTypes.string.isRequired,
};
