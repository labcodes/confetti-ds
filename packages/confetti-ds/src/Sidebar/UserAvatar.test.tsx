import React from "react";
import renderer from "react-test-renderer";

import UserAvatar from "./UserAvatar";

describe("UserAvatar", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<UserAvatar avatarSrc="src" altText="text" caption="caption" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
