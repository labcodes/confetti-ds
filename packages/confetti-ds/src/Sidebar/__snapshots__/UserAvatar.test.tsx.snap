// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar renders with base props 1`] = `
<div
  className="lab-narrow-sidebar__avatar"
>
  <div
    className="lab-narrow-sidebar__avatar-photo"
  >
    <img
      alt="text"
      src="src"
    />
  </div>
  <div
    className="lab-narrow-sidebar__avatar-text"
  >
    <span>
      text
    </span>
    <span
      className="lab-narrow-sidebar__avatar-caption"
    >
      caption
    </span>
  </div>
</div>
`;
